### ISSUE TITLE
sass-lint fails with fatal error with var/func use inside string value

### ISSUE THREAD

**Issue by viveleroi** [sasstools/sass-lint#1197]
Using sass-lint 1.12.1 (cli). I have a mixin which accepts a color argument and inserts it into an inline SVG string. sass-lint fails when it encounters this method in the inline svg.

At its simplest, this fails:

`background-image: url('#{str-replace(inspect($color), '#', '%23')}');`

```
test.scss
  11:1  error  Please check validity of the block starting from line #11  Fatal
```
The following is the real mixin I'm using and the str-replace function it depends on.

```scss
// SASS String Replace. @author Hugo Giraudel
@function str-replace($string, $search, $replace: '') {
    $index: str-index($string, $search);
    @if $index {
        @return str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace);
    }
    @return $string;
}

@mixin icon-add-knockout($color: #363E47) {
    background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20id%3D%22Layer_1%22%20data-name%3D%22Layer%201%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2014%2014%22%3E%3Cpath%20fill%3D%22#{str-replace(inspect($color), '#', '%23')}%22%20d%3D%22M11.9492%2C2.05005a6.99993%2C6.99993%2C0%2C1%2C0%2C0%2C9.9004A6.99917%2C6.99917%2C0%2C0%2C0%2C11.9492%2C2.05005ZM11%2C6.24992V7.7508a0.25067%2C0.25067%2C0%2C0%2C1-.25.25l-2.5-.0005a0.25065%2C0.25065%2C0%2C0%2C0-.25.25l0.001%2C2.50052a0.25051%2C0.25051%2C0%2C0%2C1-.25.24993l-1.501-.001A0.25094%2C0.25094%2C0%2C0%2C1%2C6%2C10.74958V8.25025a0.25073%2C0.25073%2C0%2C0%2C0-.25-0.25H3.25a0.25073%2C0.25073%2C0%2C0%2C1-.25-0.25V6.2497a0.25067%2C0.25067%2C0%2C0%2C1%2C.25-0.25l2.5%2C0.0005A0.25065%2C0.25065%2C0%2C0%2C0%2C6%2C5.7502L5.999%2C3.24975a0.25058%2C0.25058%2C0%2C0%2C1%2C.25007-0.25l1.5011%2C0.001A0.25077%2C0.25077%2C0%2C0%2C1%2C8%2C3.25075V5.75027a0.25071%2C0.25071%2C0%2C0%2C0%2C.25.25L10.75008%2C6A0.2499%2C0.2499%2C0%2C0%2C1%2C11%2C6.24992Z%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E');
    height: 14px;
    width: 14px;
}
```
